chain/network-primitives/src/time.rs

```diff
@@ -0,0 +1,98 @@
+use once_cell::sync::Lazy;
+use parking_lot::RwLock;
+use std::sync::Arc;
+pub use time::error;
+
+// TODO: consider wrapping these types to prevent interactions
+// with other time libraries, especially the the direct access
+// to the realtime (i.e. not through the Clock).
+pub type Instant = time::Instant;
+pub type Utc = time::OffsetDateTime;
+pub type Duration = time::Duration;
+
+// Instant doesn't have a deterministic contructor,
+// however since Instant is not convertible to an unix timestamp,
+// we can snapshot Instant::now() once and treat it as a constant.
+// All observable effects will be then deterministic.
+static FAKE_CLOCK_MONO_START: Lazy<Instant> = Lazy::new(Instant::now);
+
+// An arbitrary non-trivial deterministic Utc timestamp.
+const FAKE_CLOCK_UTC_START: Lazy<Utc> = Lazy::new(|| Utc::from_unix_timestamp(89108233).unwrap());
+
+#[derive(Clone)]
+enum ClockInner {
+    Real,
+    Fake(FakeClock),
+}
+
+/// Clock encapsulates a system clock, allowing to replace it
+/// with a fake in tests.
+/// Since system clock is a source of external information,
+/// it has to be replaced with a fake double, if we want our
+/// tests to be deterministic.
+///
+/// This is a reimplementation of primitives/src/time.rs
+/// with a more systematic approach.
+/// TODO: add tests, put it is some reusable package and use
+/// throughout the nearcore codebase.
+#[derive(Clone)]
+pub struct Clock(ClockInner);
+
+impl Clock {
+    pub fn real() -> Clock {
+        Clock(ClockInner::Real)
+    }
+    /// current time according to the monotonic clock
```

Comment:
```suggestion
    /// Current time according to the monotonic clock.
```

super weak opinion, but "proper sentences" is the most consistent guidellines fro comments :)

Reply:
It is not a sentence, because it doesn't have a verb. Unless by a "proper sentence" you mean "something starting with a capital letter and ending with a full stop".

Reply:
done